concepts/eventing/backend-crd.md

```diff
@@ -0,0 +1,148 @@
+    status: "True"
+    type: Subscription Controller Ready
+  eventingReady: true
+```
+
+> **Note:** As a result of having a flexible design for the backend configuration in the CRD, we decided not to mention which configuration needs to move from the Eventing controller charts to the backend CR and left it to the implementation phase.
+## Metadata
+
+The `namespaces` and `name` can be set by users. This allows to run multiple different instances of the same backend type.
+
+```yaml
+metadata:
+  name: any
+  namespace: any
+  labels:
+    kyma-project.io/eventing: backend
+```
+
+## Spec
+
+Each backend instance is defined with its own `spec`. The `spec` contains the objects `type` and `config`. While `type` accepts the string of the name of the backend (`"nats"`, `"beb"`, ...etc), the `config` accepts the key/value pairs of the actual configuration for the specified backend.
+
+
+Example how to configure `BEB` as the Eventing backend:
+
+```yaml
+spec:
+  type: "beb"
+  config:
+  - secretRef: "some-secret"
+```
+
+## Validation
+
+- Spec config validation can be done by writing a custom `ValidatingAdmissionWebhook`.
+- Defaulting the required but not-set spec config can be done by writing a custom `MutatingAdmissionWebhook`.
+
+The following scenarios should return a `validationError` and result into the backend not being created. 
+- Spec config keys that do not belong to the specified backend.
+- Spec config values set to forbidden data.
+- Spec config values using the wrong data type for the given key.
+Similar, changing the specs of an existing backend in an invalid way should be rejected and return a `validationError`.
+
+## Status
+
+The `status` contains the following fields:
+- `backendType` contains the type of the backend which is provisioned or used.
+- `conditions` contains the readiness of all the components which the backend depends on.
+- `eventingReady` is evaluated by `ANDing` all the `status.conditions`.
+
+Example of a backend CR with no validation errors:
+
+```yaml
+status:
+  backendType: nats
+  conditions:
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Backend ready
+    status: "True"
+    type: Backend Ready
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Publisher proxy deployment ready
+    status: "True"
+    type: Publisher Proxy Ready
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Subscription controller started
+    status: "True"
+    type: Subscription Controller Ready
+  eventingReady: true
+```
+
+> **Note:** The `Backend Ready` condition should indicate if the configured backend type in the spec is ready for use or not. It can be as simple as checking the underlying connection status with the backend. But that decision is left to the implementation phase.
+## Discovery
+
+Eventing users have the ability to discover the supported backends and their available configurations by doing get requests to the following endpoints:
```

Comment:
What exactly will the user be able to discover ? 
1. The list of all **implemented** eventing backends ? 
2. Or the list of all eventing backends supported in this particular cluster (e.g. there could be a nats or a beb module in the future and only one of these modules is installed).

An alternative to discovering the supported (**implemented**) backend would be an enum on the CRD openAPIV3Schema. This way users could also discover all the allowed options. 
An alternative for discovering the supported configuration would be via documentation.

Reply:
Is the API on the eventing-controller a requirement ? @D064028 
It feels a bit like we are trying to reimplement the openAPIV3Schema validation 🤔

Reply:
Discovery section was changed. New version is here: https://github.com/kyma-project/community/pull/671/files